frontend/packages/dev-console/src/components/pipelineruns/logs/MultiStreamLogs.tsx

```diff
@@ -0,0 +1,104 @@
+import { containerToLogSourceStatus } from '../../../utils/pipeline-utils';
+import { getRenderContainers } from './logs-utils';
+import Logs from './Logs';
+import './MultiStreamLogs.scss';
+
+type MultiStreamLogsProps = {
+  resource: PodKind;
+  taskName: string;
+};
+
+export const MultiStreamLogs: React.FC<MultiStreamLogsProps> = ({ resource, taskName }) => {
+  const scrollPane = React.useRef<HTMLDivElement>();
+  const completedRef = React.useRef<boolean[]>([]);
+  const [renderToCount, setRenderToCount] = React.useState(0);
+  const [isFullscreen, fullscreenRef, fullscreenToggle] = useFullscreen<HTMLDivElement>();
+  const [scrollDirection, handleScrollCallback] = useScrollDirection();
+  const [autoScroll, setAutoScroll] = React.useState(true);
+  const { containers, stillFetching } = getRenderContainers(resource);
+  const dataRef = React.useRef<ContainerSpec[]>(null);
+  dataRef.current = containers;
+
+  const handleComplete = React.useCallback((containerName) => {
+    const index = dataRef.current.findIndex(({ name }) => name === containerName);
+    completedRef.current[index] = true;
+    const newRenderTo = dataRef.current.findIndex((c, i) => completedRef.current[i] !== true);
+    if (newRenderTo === -1) {
+      setRenderToCount(dataRef.current.length);
+    } else {
+      setRenderToCount(newRenderTo);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (!scrollDirection) return;
+    if (scrollDirection === ScrollDirection.scrollingUp && autoScroll === true) {
+      setAutoScroll(false);
+    }
+    if (scrollDirection === ScrollDirection.scrolledToBottom && autoScroll === false) {
+      setAutoScroll(true);
+    }
+  }, [autoScroll, scrollDirection]);
+  const downloadLogs = () => {
+    if (!scrollPane.current) return;
+    const logString = scrollPane.current.innerText;
+    const blob = new Blob([logString], {
+      type: 'text/plain;charset=utf-8',
+    });
+    saveAs(blob, `${taskName}.log`);
+  };
+
+  const containerStatus: ContainerStatus[] = resource.status?.containerStatuses ?? [];
+  return (
+    <div
+      ref={fullscreenRef}
+      className={classNames('odc-multi-stream-logs', {
+        'odc-multi-stream-logs--fullscreen': isFullscreen,
+      })}
+    >
+      <LogControls
+        isFullscreen={isFullscreen}
+        onDownload={downloadLogs}
+        toggleFullscreen={fullscreenToggle}
+      />
+      <div
+        className={classNames('odc-multi-stream-logs__container', {
+          'is-edge': window.navigator.userAgent.includes('Edge'),
+        })}
+        ref={scrollPane}
+        onScroll={handleScrollCallback}
+        data-test-id="logs-task-container"
+      >
+        <div className="odc-multi-stream-logs__container--pos">
+          <div className="odc-multi-stream-logs__taskName">
+            {taskName}
+            {stillFetching && (
+              <span className="odc-multi-stream-logs__taskName--loading">
```

Comment:
Same here, don't think the modifier is needed as this is just a conditional element.
```suggestion
              <span className="odc-multi-stream-logs__loadingIndicator">
```
(for instance)